File: MSRpdf_cop_Clayton/MSRpdf_cop_Clayton.R
# ---------------------------------------------------------------------
# Book:         
# ---------------------------------------------------------------------
# Quantlet:     MSRpdf_cop_Clayton
# ---------------------------------------------------------------------
# Description:  MSRpdf_cop_Clayton plots the Clayton copula density 
#		        with parameter p=2.
# ---------------------------------------------------------------------
# Usage:        MSRpdf_cop_Clayton
# ---------------------------------------------------------------------
# Inputs:       None
# ---------------------------------------------------------------------
# Output:       Plot of the Clayton copula density.
# ---------------------------------------------------------------------
# Example:     
# ---------------------------------------------------------------------
# Author:       Zografia Anastasiadou, 20110603
# ---------------------------------------------------------------------

rm(list=ls(all=TRUE))
#setwd("C:/...")

install.packages("lattice")
library(lattice)

u<-seq(0,1,by=0.02)
v=u
m=length(u) 
n=length(v)
U=matrix(rep(u,each=n),nrow=n)
V=matrix(rep(v,m),nrow=n)

p=2
gcopuly=(U^(-p)+V^(-p)-1)^(-2-1/p)*(U*V)^(-p-1)*(p+1)

s=expand.grid(u=u,v=v)
wireframe(gcopuly~u*v,s,shade=TRUE,xlab="X",ylab="Y",zlab="Z",main="Clayton Copula Density, theta = 2",scales=list(arrows=FALSE))

File: MSRpdf_cop_Clayton/MSRpdf_cop_Clayton.m

function MSRpdf_cop_Clayton
[u,v]   = meshgrid(0:0.02:1);
p       = 2;
gcopuly = (u.^(-p) + v.^(-p)-1).^(-2-1/p).*(u.*v).^(-p-1).*(p+1);
grid on
mesh(u, v, gcopuly)
xlabel('X');
ylabel('Y');
zlabel('Z');
title('Clayton Copula Density, \\theta = 2')